src/core/DesktopManager.ts

```diff
@@ -175,7 +175,19 @@ export default class implements Publisher<DesktopEvent>, GarbageCollector {
   moveToMonitor(target: Meta.Window, monitorIdx?: number) {
     monitorIdx = monitorIdx ?? (target.get_monitor() + 1) % this.monitors.length;
     target.unmaximize(Meta.MaximizeFlags.BOTH);
-    target.move_to_monitor(monitorIdx);
+    if (monitorIdx === target.get_monitor()) {
```

Comment:
I suppose this is what this semantically expresses? Maybe put it on top of the function so that it is clear that the function hasn't have any effect for single-monitor setups.

```suggestion
    if (this.monitory.length === 1) {
```